content/advent-2015/gopy.md

```diff
@@ -0,0 +1,342 @@
++++
+author = ["Sebastien Binet"]
+date = "2015-12-06T16:41:54+01:00"
+linktitle = "Gopy"
+series = ["Advent 2015"]
+title = "gopy: extending CPython with Go"
+
++++
+
+Applications and libraries do not live in a vacuum.
+This is of course true for `Go` packages and commands.
+More often than not, you need your code to interact with legacy applications or
+old and battle tested libraries.
+
+The standard library of `Go` provides many facilities for interacting with such
+entities: `encoding/json`, `encoding/xml` or `net/rpc`.
+Reach out of the standard library and you have `ProtoBuf`.
+All of these facilities meant to arrange for some kind of IPC (Inter-Program
+Communication) and before `Go-1.5`, this was the only portable way to do so.
+
+
+With `Go-1.5` and the new [execution
+modes](https://docs.google.com/document/d/1nr-TQHw_er6GOQRsF6T43GGhFDelrAP0NqSS_00RgZQ/edit)
+it is now possible to create `C` shared libraries from a `Go` package: that's
+performed via the `-buildmode=c-shared` command-line option.
+As you have seen in [libc-hooking-go-shared-libraries](https://blog.gopheracademy.com/advent-2015/libc-hooking-go-shared-libraries/), creating a `C` shared library is not particularly complicated but the process does involve a few steps.
+
+[gopy](https://github.com/go-python/gopy) automates the drudgery work of
+creating a `CPython` C-extension module out of a `Go` package: eventually
+allowing you to write nice concurrent libraries in `Go` and share them with your
+`CPython` friends.
+Given a `Go` package, `gopy` will:
+
+- inspects the `Go` package
+- extracts the exported `types`, `funcs`, `vars` and `consts`
+- creates a `Go` package that `cgo` exports the exported entities
+- creates a `C` extension module, using the `CPython` API, calling these `cgo` exported entities
+- compiles everything together into a `.so` shared object
+
+## Installation
+
+`gopy` is a pure-`Go` command and can thus be installed like so:
+
+```sh
+sh> go get github.com/go-python/gopy
+```
+
+Of course, at runtime, for `gopy` to be able to actually generate the `CPython`
+extension module, it will need:
+
+- the `python-dev` package (which contains the `CPython` development headers and
+  `libpython.so.2.X`)
```

Comment:
this made me wonder if it supports Python 3? It might be a good idea to mention the supported Python versions near the beginning of the article.